src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/AnnotatedVariantProducer.java

```diff
@@ -22,9 +22,36 @@
  * Given identified pair of breakpoints for a simple SV and its supportive evidence, i.e. chimeric alignments,
  * produce an annotated {@link VariantContext}.
  */
-class AnnotatedVariantProducer implements Serializable {
+public class AnnotatedVariantProducer implements Serializable {
     private static final long serialVersionUID = 1L;
 
+    public static Iterator<VariantContext> produceMultipleAnnotatedVcFromNovelAdjacency(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations,
+                                                                                        final Iterable<SvType> inferredType,
+                                                                                        final Iterable<ChimericAlignment> contigAlignments,
+                                                                                        final Broadcast<ReferenceMultiSource> broadcastReference)
+            throws IOException {
+
+        Utils.validateArg(inferredType.iterator().hasNext(),
+                "Input novel adjacency doesn't have any inferred type: \n" +
+                        Utils.stream(contigAlignments).map(ChimericAlignment::onErrStringRep).collect(Collectors.toList()));
+
+        final Iterator<SvType> it = inferredType.iterator();
+        final VariantContext record =
+                produceAnnotatedVcFromInferredTypeAndRefLocations(novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc, -1,
+                        novelAdjacencyReferenceLocations.complication, it.next(), contigAlignments, broadcastReference);
+
+        final List<VariantContext> result = new ArrayList<>();
+        result.add(record);
+        // hack for now because up to this point inferredType would have max of 2 only
+        while (it.hasNext()) {
+            final VariantContext mateRecord =
+                    produceAnnotatedVcFromInferredTypeAndRefLocations(novelAdjacencyReferenceLocations.leftJustifiedRightRefLoc, -1,
+                            novelAdjacencyReferenceLocations.complication, it.next(), contigAlignments, broadcastReference);
+            result.add(mateRecord);
+        }
+        return result.iterator();
```

Comment:
Why not return `List<>` here? A caller can easily convert it to an iterator if needed. Returning `List` provides more flexibility to the caller.